templates.rst

```diff
@@ -0,0 +1,1061 @@
+.. index::
+   single: Templating
+
+Creating and Using Templates
+============================
+
+A template is a text file that can generate any text-based format (HTML, XML,
+CSV, etc.). For example, in a web application that displays some news, you can
+create a text file (e.g. ``news.php``) with the HTML structure and PHP logic of
+all news pages. Then, a :doc:`Symfony controller </controllers>` can *render*
+that ``news.php`` template passing the contents of each news to generate the
+resulting HTML contents sent to the user.
+
+Templates were traditionally created with PHP, but in Symfony applications they
+are created with Twig, a flexible, fast, and secure template engine.
```

Comment:
I think we can shorten all of this (and yes, I probably wrote at least part of this longer version originally - but times have changed):

```rst
A template is *the* best way to organize & render HTML from inside your application, whether you need to
render HTML from a :doc:`controller </controllers>` or generate the :doc:`contents of an email </mailer>`.
Templates in Symfony are created with Twig: a flexible, fast, and secure template engine.
```